signac/common/config.py

```diff
@@ -20,65 +20,64 @@ def _get_project_config_fn(root):
     return os.path.abspath(os.path.join(root, PROJECT_CONFIG_FN))
 
 
-def _get_config_dirname(fn):
-    # We could use pathlib's `.paths` attribute to remove the trailing project
-    # config filename, but that has significant performance
-    # implications: this takes ~200 ns, whereas
-    # os.path.join(*(pathlib.PosixPath(fn).parts[:-2])) takes ~6 us, almost a
-    # 20x slowdown.
-    return fn.replace(os.sep + PROJECT_CONFIG_FN, "")
+def _contains_config_file(root):
+    """Determine if the root directory contains a signac configuration file.
 
+    Parameters
+    ----------
+    root : str
+        Path to search. Uses ``os.getcwd()`` if None (Default value = None).
 
-def _search_local(root):
+    Returns
+    --------
+    str or None
+        The path to the configuration file if one is found, otherwise None.
+    """
     fn_ = _get_project_config_fn(root)
     if os.path.isfile(fn_):
-        yield fn_
+        return root
+    return None
 
 
-def _search_tree(root=None):
-    """Locates signac configuration files in a directory hierarchy.
+def _locate_config_dir(root):
+    """Locates root directory containing a signac configuration file in a directory hierarchy.
 
     Parameters
     ----------
     root : str
-        Path to search. Uses ``os.getcwd()`` if None (Default value = None).
+        Starting path to search.
 
+    Returns
+    --------
+    str or None
+        The root directory containing the configuration file if one is found, otherwise None.
     """
-    if root is None:
-        root = os.getcwd()
     while True:
-        yield from _search_local(root)
+        if _contains_config_file(root):
+            return root
         up = os.path.abspath(os.path.join(root, ".."))
         if up == root:
             logger.debug("Reached filesystem root, no config found.")
-            return
+            return None
         else:
             root = up
 
 
-def _search_standard_dirs():
-    """Locates signac configuration files in standard directories."""
-    # For now this search only finds user-specific files, but it could be
-    # updated in the future to support e.g. system-wide config files.
-    for fn in (USER_CONFIG_FN,):
-        if os.path.isfile(fn):
-            yield fn
-
-
 def read_config_file(filename, configspec=None, *args, **kwargs):
     """Read a configuration file.
 
-    Parameters
-    ----------
-    filename : str
-        The path to the file to read.
-    configspec : List[str], optional
-        The key-value pairs supported in the config.
+        Parameters
```

Comment:
It looks like an extra indentation was added to the next few lines. The return section was also messed up?